replica/table.cc

```diff
@@ -739,21 +739,18 @@ future<> table::maybe_wait_for_sstable_count_reduction() {
     if (_async_gate.is_closed() || !_compaction_manager.can_perform_regular_compaction(this)) {
         co_return;
     }
-    auto num_runs_with_memtable_origin = [this] {
+    auto num_runs_for_compaction = [this] {
         auto& cm = _compaction_manager;
         auto& cs = get_compaction_strategy();
         auto desc = cs.get_sstables_for_compaction(as_table_state(), cm.get_strategy_control(), cm.get_candidates(*this));
         return boost::copy_range<std::unordered_set<utils::UUID>>(
             desc.sstables
-            | boost::adaptors::filtered([] (const sstables::shared_sstable& sst) {
-                return sst->get_origin() == "memtable";
-            })
```

Comment:
That's a big change :-) But I think it's the right thing to do. If any bucket has severely fallen behind, then it means read amplification is terrible and LSM tree needs to stop ingesting more data, so compaction can keep up and maintain read amplification within bounds. Compaction strategy should strive to return the bucket which is causing most of the backlog, if any strategy isn't doing that, then it is its fault. 

LCS returns L0 if it's falling behind
STCS and ICS returns any bucket with most files
So does TWCS, which is essentially a set of LSM trees.